hicexplorer/hicBuildMatrix.py

```diff
@@ -822,74 +844,363 @@ def main(args=None):
 
         for mate in [mate1, mate2]:
             # fill in coverage vector
-            vec_start = max(0, mate.pos - mate_bin.begin) / binsize
-            vec_end = min(len(coverage[mate_bin_id]), vec_start +
-                          len(mate.seq) / binsize)
-            coverage[mate_bin_id][vec_start:vec_end] += 1
+            vec_start = max(0, mate.pos - mate_bin.begin) / pBinsize
+            length_coverage = pCoverageIndex[mate_bin_id].end - pCoverageIndex[mate_bin_id].begin
+            vec_end = min(length_coverage, vec_start +
+                          len(mate.seq) / pBinsize)
+            coverage_index = pCoverageIndex[mate_bin_id].begin + vec_start
+            coverage_end = pCoverageIndex[mate_bin_id].begin + vec_end
+            for i in xrange(coverage_index, coverage_end, 1):
```

Comment:
I am not sure what to do with xrange vs. range. I know it is a potential issue to port the source code to Python 3 but on the other hand the behavior of 'range' in Python 2.7 is not wanted.